lib/DocumentRetriever.js

```diff
@@ -67,32 +67,48 @@ function main(documentRetrieverTypeString) {
 					return res;
 				}, {});
 				const index = utils.array_flatten(Object.values(indexes)).find((index) => {
-					const {hash, range} = index.KeySchema.reduce((res, item) => {
+					const {hash/*, range*/} = index.KeySchema.reduce((res, item) => {
 						res[item.KeyType.toLowerCase()] = item.AttributeName;
 						return res;
 					}, {});
-					return (comparisonChart[hash] || {}).type === "EQ" && (!range || !comparisonChart[range] || comparisonChart[range].type === "EQ");
+					// TODO: we need to write logic here to prioritize indexes with a range key that is being queried.
```

Comment:
Is this still up to date?

Reply:
@MickL Yes it is. Gonna maybe do that in a separate PR. That issue is slightly lower priority compared to the rest of this PR. So I want to get this PR through first then I can come back to that in a cleanup PR.